File: module2/src/case_study/models/Customer.java
package case_study.models;

public class Customer {
    private String name;
    private String birthday;
    private String gender;
    private String idNumber;
    private String phoneNumber;
    private String email;
    private String customerType;
    private String address;
    private Service service = new Villa("N/A", "N/A");

    public Customer() {
    }

    // Constructor has attribute service <Services>
    public Customer(String name, String birthday, String gender, String idNumber, String phoneNumber,
                    String email, String customerType, String address,
                    Service service) {
        this(name, birthday, gender, idNumber, phoneNumber,
                email, customerType, address);
        this.service = service;
    }

    // Constructor DON'T have attribute service
    public Customer(String name, String birthday, String gender, String idNumber, String phoneNumber,
                    String email, String customerType, String address) {
        this.name = name;
        this.birthday = birthday;
        this.gender = gender;
        this.idNumber = idNumber;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.customerType = customerType;
        this.address = address;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getIdNumber() {
        return idNumber;
    }

    public void setIdNumber(String idNumber) {
        this.idNumber = idNumber;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getCustomerType() {
        return customerType;
    }

    public void setCustomerType(String customerType) {
        this.customerType = customerType;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Service getService() {
        return service;
    }

    public void setService(Service service) {
        this.service = service;
    }

    public void showInfo() {
        System.out.printf("%-25s%-16s%-12s%-14s%-18s%-20s%-18s%-20s%-15s%s\n",
                name, birthday, gender, idNumber, phoneNumber, email, customerType, address,
                this.getService().getId(), this.getService().getServiceName());
    }
}

File: module5/case-study/src/app/model/customer.ts
export class Customer{
  private _name: string;
  private _birthday: string;
  private _idCard: string;
  private _phone: string;
  private _email: string;
  private _address: string;
  private _customerType: string;

  constructor(name: string, birthday: string, idCard: string, phone: string, email: string, address: string, customerType: string){
    this._name = name;
    this._birthday = birthday;
    this._idCard = idCard;
    this._phone = phone;
    this._email = email;
    this._address = address;
    this._customerType = customerType;
  }


  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get birthday(): string {
    return this._birthday;
  }

  set birthday(value: string) {
    this._birthday = value;
  }

  get idCard(): string {
    return this._idCard;
  }

  set idCard(value: string) {
    this._idCard = value;
  }

  get phone(): string {
    return this._phone;
  }

  set phone(value: string) {
    this._phone = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }

  get address(): string {
    return this._address;
  }

  set address(value: string) {
    this._address = value;
  }

  get customerType(): string {
    return this._customerType;
  }

  set customerType(value: string) {
    this._customerType = value;
  }
}
